### PR TITLE
bpo-30727: Fix a race condition in test_threading.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

**Comment by vstinner** [comment]
I would prefer to change the test in the master branch, and backport it later.

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by serhiy-storchaka** [comment]
> I would prefer to change the test in the master branch, and backport it later.

Done.

File: Lib/test/lock_tests.py [hunk 1]
```diff
@@ -461,21 +461,28 @@ def _check_notify(self, cond):
...
             cond.release()
             results2.append((result, phase_num))
         b = Bunch(f, N)
         b.wait_for_started()
-        _wait()
+        # first wait, to ensure all workers settle into cond.wait() before
+        # we continue. See issues #8799 and #30727.
+        while len(ready) < 5:
+            _wait()
+        ready = []
```
Comment (vstinner) [inline Lib/test/lock_tests.py]: I suggest ready.clear() rather than ready=[], just in case.

Comment (serhiy-storchaka) [reply Lib/test/lock_tests.py]: Actually this doesn't matter, **all** workers settle into `cond.wait()`.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**